src/sentry/performance_issues/detectors/large_payload_detector.py

```diff
@@ -23,15 +25,40 @@
 EXTENSION_REGEX = re.compile(r"\.([a-zA-Z0-9]+)/?(?!/)(\?.*)?$")
 EXTENSION_ALLOW_LIST = ("JSON",)
 
+# Content-Type values that typically indicate file downloads
+# Focused on the most common scenarios that would cause false positives
+FILE_DOWNLOAD_CONTENT_TYPES = {
+    "application/octet-stream",  # Generic binary file
+    "application/pdf",  # PDF documents
+    "application/zip",  # ZIP archives
+    "image/",  # All image types
+    "video/",  # All video types
+    "audio/",  # All audio types
+}
+
 
 class LargeHTTPPayloadDetector(PerformanceDetector):
+    """
+    Detects large HTTP payloads that may indicate performance issues.
+
+    This detector excludes file downloads based on:
+    1. File extensions in URLs (existing behavior)
+    2. HTTP response headers indicating file downloads (new behavior):
+       - Content-Disposition: attachment
+       - Content-Type values indicating file downloads (e.g., application/pdf, image/*, video/*, etc.)
+
+    This addresses the issue where file download endpoints using internal unique identifiers
+    (without file extensions in URLs) were incorrectly flagged as performance issues.
+    """
+
     type = DetectorType.LARGE_HTTP_PAYLOAD
     settings_key = DetectorType.LARGE_HTTP_PAYLOAD
 
     def __init__(self, settings: dict[DetectorType, Any], event: dict[str, Any]) -> None:
         super().__init__(settings, event)
 
         self.consecutive_http_spans: list[Span] = []
+        self.organization = self.settings.get("organization")
```

Comment:
need organization for feature flag check but previous did not have access to it - will remove this workaround when we remove the flag